src/payment/strategies/adyenv2/adyenv2-initialize-options.ts

```diff
@@ -26,7 +26,11 @@ export default interface AdyenV2PaymentInitializeOptions {
     options?: Omit<CreditCardComponentOptions, 'onChange'>;
 
     /**
-     * Optional. Contains all three ds 2 options
+     * Specify Three3DSChallenge Widget Size
      */
-    threeDS2Options?: ThreeDS2ComponentOptions;
```

Comment:
I don't think you can make this change anymore as this would be a breaking change. That is why you should never ever merge WIP PRs into master. Also, ideally, the name of the interface should be prefixed with `Adyen` because in the end all these interfaces are exported under a single TS interface file (https://github.com/bigcommerce/checkout-sdk-js/blob/master/dist/checkout-sdk.d.ts). This might cause a name collision in the future. So I suggest you mark `ThreeDS2ComponentOptions` as deprecated and create another one called `AdyenV2ThreeDS2ComponentOptions` and declare `threeDS2Options` field as `AdyenV2ThreeDS2ComponentOptions | ThreeDS2ComponentOptions`. This comment also applies to `CreditCardComponentOptions `

Reply:
Hi David, thank you for your concern. I believe the experiment hasn't been rolled out yet and dependencies such BigPay have not yet been merged either.

Reply:
Not 100% sure what you meant by "Adyen's feedback" because the API of the SDK and the API of Adyen JS client are two separate things. We design how the public interface should look like for the SDK. And the consumer of the SDK would provide values according to that interface. Similarly, Adyen designs how their public interface should look like for their client library. And the SDK, as the consumer of the client library, would provide values according to their interface. Sometimes the interfaces might be the same, sometimes they might not, which means the values need to be mapped.

You could essentially do the same thing in a backward compatible way, i.e.:

```ts
threeDS2Options: {
    widgetSize?: string;
    onLoading?(cancel: () => void): void;
    onComplete?(): void;
}
```

Regarding the use of experiments, imagine this: version 1.2.3 contains your WIP change which you think it wouldn't matter because the WIP change wouldn't be executed unless you turn on the experiment. Then one day everything is ready and you turn on the experiment. That would effectively turn on the WIP change, because although our checkout is always on the latest version of SDK, other third party checkouts might still be using version 1.2.3. That is why you should never merge a WIP PR and you should never change the interface of a method after it is merged to master.